docs/testing/unit-testing.md

```diff
@@ -1070,3 +1070,5 @@ Jest incorporates most of the features of Sinon with a slimmer API:
 - `jest.fn()` can be used in place of `sinon.stub()`.
 - `jest.spyOn(object, method)` can be used in place of `sinon.spy(object, method)` or `sinon.stub(object, method)` (with the caveat that the method being spied upon will still be called by default).
 - `jest.useFakeTimers()` can be used in place of `sinon.useFakeTimers()` (though note that Jest's "clock" object had fewer features than Sinon's prior to Jest v29.5).
+
+**Avoid jest manual mocks:** According to jest `Manual mocks are defined by writing a module in a __mocks__/ subdirectory immediately`. These types of mocks are automatically picked up by jest for all tests. We should be very careful when writing this types of mocks as they will be shared across all tests. And with UI integration tests, as we are aiming to mock as little as possible, we should avoid this at all costs.
```

Comment:
Hmm, I see why you included this here, but maybe we should rename this section from "Use Jest's mock functions instead of Sinon" to "Use Jest's API instead of Sinon's"?

Additionally, to make this guideline stand out and make it linkable on its own, should we move it to its own header? Finally, since this is the document on unit testing, do we need to include anything around integration testing here?

``` suggestion
## Avoid manual mocks

According to Jest's documentation, "Manual mocks are defined by writing a module in a `__mocks__/` subdirectory immediately". These types of mocks are automatically picked up by Jest for all tests. We should be very careful when writing these types of mocks as they will be shared across all tests.
```